### ISSUE TITLE
InstrumentationContext onDispatched not called for parsing

### ISSUE THREAD

**Issue by Keisuki** [graphql-java/graphql-java#1620]
**Describe the bug**
For a couple of the methods on Instrumentation (beginParse and beginValidation) the onDispatched method on the returned InstrumentationContext is not called.  

**To Reproduce**
My Instrumentation, which is later passed into a GraphQL instance:
```
public class TestInstrumentation extends SimpleInstrumentation {
  @Override
  public InstrumentationContext<Document> beginParse(InstrumentationExecutionParameters parameters) {
    return new InstrumentationContext<>() {
      @Override
      public void onDispatched(CompletableFuture<Document> result) {
        System.out.println("Dispatched");
      }

      @Override
      public void onCompleted(Document result, Throwable t) {
        System.out.println("Completed");
      }
    };
  }
}
```

Expected:
```
Dispatched
Completed
```

Actual:
```
Completed
```

**Comment by Keisuki** [comment]
This appears to apply to beginExecution also.

**Comment by bbakerman** [comment]
Yeah this is an oversight because the actions here are not asynchronous.  That is they have no clear dispatch and later completion.

I agree for consistency they should happen even if its happening synchronously.

As a workaround I guess you should do you call back work in the onCompleted callback until we get this fixed.

**Comment by Keisuki** [comment]
For now, I am calling onDispatch before returning the InstrumentationContext from the Instrumentation, as I don't need the future.  That way will be easiest to modify back into how it should be, should this be fixed.

**Comment by dondonz** [comment]
Fixed by #2730
